src/tests/test_console.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+import sys
+
+sys.path.append(str(Path(__file__).parents[2] / "src" / "pyodide-py"))
+
+from pyodide import console  # noqa: E402
+
+
+def test_stream_redirection():
+    my_buffer = ""
+
+    def stdout_callback(string):
+        nonlocal my_buffer
+        my_buffer += string
+
+    my_stream = console._StdStream(stdout_callback)
+
+    print("foo", file=my_stream)
+    assert my_buffer == "foo\n"
+    print("bar", file=my_stream)
+    assert my_buffer == "foo\nbar\n"
+
+
+def test_interactive_console_streams():
+
+    my_stdout = ""
+    my_stderr = ""
+
+    def stdout_callback(string):
+        nonlocal my_stdout
+        my_stdout += string
+
+    def stderr_callback(string):
+        nonlocal my_stderr
+        my_stderr += string
+
+    shell = console.InteractiveConsole(
+        stdout_callback=stdout_callback, stderr_callback=stderr_callback
+    )
+
+    # std names
+    assert sys.stdout.name == "<stdout>"
+    assert sys.stderr.name == "<stderr>"
+
+    # std redirections
+    print("foo")
+    assert my_stdout == "foo\n"
+    print("bar", file=sys.stderr)
+    assert my_stderr == "bar\n"
+
+    # redirections disabled at destruction
+    shell.restore_stdstreams()
```

Comment:
If this test fails before this line does it mean that we get monkeypatched stdout/stderr globally? 

Maybe it would be good add a pytest fixure that stores and restores the stdout just in case. Something like,
```py
@pytest.fixture
def restore_stdstreams():
    # save streams to local variables
    yield
    # restore streams after the test function exits

def test_interactive_console_streams(restore_stdstreams):
    ...

```

BTW, it's interesting that pytest also captures stderr/stdout but does it at the level of file system descriptors https://docs.pytest.org/en/stable/capture.html#setting-capturing-methods-or-disabling-capturing

Reply:
Right, it could be a bad issue!